### DISCUSSION TITLE
Adding security to a nginx-proxy docker-compose setup

### DISCUSSION THREAD

**Discussion by r0ckyjoe** [nginx-proxy/nginx-proxy#2160] [Q&A] [answered]
Hello team,
I've just finished to setup a docker-compose.yml file with nginx, docker-gen and some web services configured as follows:

`++++++++++++++++++++++++++++++++++++++++`
`+    VIRTUAL_HOST     +  VIRTUAL_PORT  +`
`++++++++++++++++++++++++++++++++++++++++`
`+ app1.mydomain.local +   1800         +`
`++++++++++++++++++++++++++++++++++++++++`
`+ app2.mydomain.local +   1900         +`
`++++++++++++++++++++++++++++++++++++++++`
`+ app3.mydomain.local +   2000         +`
`++++++++++++++++++++++++++++++++++++++++`
`+ app4.mydomain.local +   2100         +`
`++++++++++++++++++++++++++++++++++++++++`
`+ app5.mydomain.local +   2200         +`
`++++++++++++++++++++++++++++++++++++++++`

Some of those are pure http apps some others can be configured to use https.
Even though the applications are not exposed to the internet, in order to get closer to a Zero-Trust setup, I would like to add a layer of security by enabling an https Front End. 
Ideally, I would like to reach them via something like s-app1.mydomain.local, s-app2.mydomain.local, s-app3.mydomain.local, ...

How can I do that?

Thanks in advance for the kind reply!

~RJ

**Comment by rhansen** [comment, marked as answer]
This seems like a straightforward setup, so [`README.md`](https://github.com/nginx-proxy/nginx-proxy/blob/main/README.md) should sufficiently explain how to use nginx-proxy for your use case.  If something is not working for you, or if the documentation is unclear, please provide more details.

**Comment by r0ckyjoe** [comment]
there must be something I miss then. I'll have a look at the nginx documentation too in order to better understand how stuff works. If I should still face issues I'll get back asking for help.
Thx!
